nucliadb/nucliadb/ingest/orm/utils.py

```diff
@@ -82,3 +82,25 @@ def set_title(writer: BrokerMessage, toprocess: PushPayload, title: str):
 
 def compute_paragraph_key(rid: str, paragraph_key: str) -> str:
     return paragraph_key.replace("N_RID", rid)
+
+
+def choose_matryoshka_dimension(dimensions: list[int]) -> int:
+    """Given a list of matryoshka embedding available dimensions, choose one to
+    set the vector dimension.
+    """
+    if len(dimensions) == 0:
+        raise ValueError("Can't choose matryoshka dimension from an empty list")
+
+    threshold = 2000
+    previous = None
+    for dimension in sorted(dimensions):
+        if dimension > threshold:
+            break
+        previous = dimension
+
+    if dimension > threshold:
+        if previous is None:
+            return dimension
+        else:
+            return previous
+    return dimension
```

Comment:
Which is the rationale for this logic?

Reply:
We are currently supporting 1.5k vector dimensions. This gets the closest value to 2k, something I think we can manage. There's some matryoshka models with 3k vector dimension we'll not use for now.

It's an heuristic though, we can try different options and make the 2k a env variable so we can tweak while running